### ISSUE TITLE
One line translation from OT to figure in FigTree or similar 

### ISSUE THREAD

**Issue by mjy** [OpenTreeOfLife/hackathon#21]
FigTree http://tree.bio.ed.ac.uk/software/figtree/ is still a popular figure generating tool.  

If I do:

```
OT2Fig.new_figure(
   file_listing_taxa_to_show: File.open('taxa_one_per_line.txt'),
   output: :figtree
 )
```

Then I get new file formatted for FigTree rendering. 

Could be extended with options etc.  Could be extended to multiple output types using R, DOT/Graphviz, etc.  Edit: closely related to #17 it seems.

**Comment by chinchliff** [comment]
Figtree reads nexus and newick files. We already output newick trees from api services. You can add lots of formatting to a figtree file for branch lengths, etc., but I'm not sure what kinds of formatting you might want? We probably won't provide the option to annotate trees with figtree formatting, but that is a potential activity for clients or users to do.

I agree this is closely related to #17. Perhaps one project could be to develop a tool that would accept:
1. An input tree (e.g. from opentree)
2. Some annotation data (e.g. from some other database)
3. Some phylostylotastic css

And would produce a tree annotated with figtree (or any tree visualization program) formatting info that could then be viewed in that program.

**Comment by mjy** [comment]
Right, so I'm thinking of a client side wrapper on the OT API that pulls
down a graph and also does X, turning X into annotations in Fig tree
format.  For example, pull down a tree, iterate through the names therein,
query BHL for presence or absence of names (and by proxy some other
information), turn that data into annotations on the tree, spit out the
annotated tree for visualization.  You can use the branch widths, colors
etc. in lots of different manners, see
http://hymao.blogspot.ca/2009/10/more-visualization-experiments.html for
example.

**Comment by chinchliff** [comment]
Cool figure! That sounds like a useful tool. Figtree definitely has lots of
options for rendering interesting things. You may want to be in touch with
@jimallman as well, who is working on a tree illustrator tool (under the OT
umbrella) that may provide another way to render annotated trees (though
it's still in fairly early stages at this point).

**Comment by arlin** [comment]
Phylowidget (http://www.phylowidget.org/) also has an URL API.  This means that it can be used to access OpenTree (or TreeBASE, etc) directly without modification.

**Comment by curtislisle** [comment]
This idea of "enriching" a tree by adding annotations to the nodes and/or edges is interesting to the Arbor team.  Arbor supports workflows where the data passing between steps can be a general graph with annotations on all the nodes and edges.  We will be interested to see how this "rich tree" with formatting, etc. develops.  If we decide on common ways to annotate, we could exchange formatted trees with annotations between packages.

**Comment by chinchliff** [comment]
The NexSON specification that opentree uses has evolved (and
stabilized) considerably, and supports arbitrary annotations on any
element, in addition to specifically defining a lot of metadata properties
that are common for trees (like branch lengths, node support, etc).
@mtholder, where is the most current specification for NexSON?

**Comment by mjy** [comment]
What services/software can _ingest_ NexSON with these annotations? Are
there existing (i.e. can play with now) APIs that "do stuff" with richly
annotated NexSON?

M

**Comment by jimallman** [comment]
> where is the most current specification for NexSON?

I believe _(corrections welcome!)_ that the "canonical" NexSON reference is this [WikiSpaces page](http://opentree.wikispaces.com/NexSON), but the version on our [GitHub wiki](https://github.com/OpenTreeOfLife/phylesystem-api/wiki/NexSON) is a bit more up-todate. 

There's also a GitHub page with [full details on NexSON annotations](https://github.com/OpenTreeOfLife/phylesystem-api/wiki/Annotations-in-NexSON).

**Comment by jimallman** [comment]
> What services/software can _ingest_ NexSON with these annotations?

I'm not sure what else is out there, but the Open Tree curation webapp reads and displays a number of common NexSON annotations. It also creates its own, for example to capture OTU mapping "hints" for each study. The code for this is rather spread out, but I can provide pointers if I know what you're looking for.

**Comment by mtholder** [comment]
I think that most of the tags used in NexSON are on [the tags page on the phylesystem-api wiki](https://github.com/OpenTreeOfLife/phylesystem-api/wiki/NexSON). That page should agree with the wikispaces pages that @jimallman linked to. I'll check on that.
The syntactic conventions (mainly of interest to those curious about how NexSON maps to NeXML) are described on [the syntax page on the phylesystem-api wiki](https://github.com/OpenTreeOfLife/phylesystem-api/wiki/HoneyBadgerFish). 
The annotations page that @jimallman  mentioned is used by tools that make comments on files (non-crucial statements about the data).

There are multiple versions of NexSON. Which can certainly lead to confusion. But the main tool that serves up NexSON content is able to return them in the version that the client wants (via a `output_nexml2json` argument (described on [the phylesystem-api docs page](https://github.com/OpenTreeOfLife/phylesystem-api/tree/master/docs#get-arguments)). So we certainly do not expect users of the API to learn about multiple versions of NexSON. For that matter, other formats can be returned by that service (see [this part of the docs](https://github.com/OpenTreeOfLife/phylesystem-api/tree/master/docs#output-conversion-of-get)).

Thus far, I think that only open tree software makes use of the tags that @chinchliff was referring to.

_edit_: closed a set of parentheses.

**Comment by daisieh** [comment]
Leaving a link here: perhaps we could make a Phylostylotastic engine for http://pythonhosted.org/ete2/tutorial/tutorial_drawing.html#rendering-trees-as-images
